build.xml

```diff
@@ -31,86 +31,44 @@
   <property name="adaptor.class" 
       value="${adaptor.pkg.name}.examples.AdaptorTemplate"/>
   <property name="adaptor.args" value=""/>
-  <property name="cobertura.dir" value="${basedir}/../cobertura/"/>
+
+  <!-- Ivy settings -->
+  <property name="ivy.install.version" value="2.4.0" />
+  <condition property="ivy.home" value="${env.IVY_HOME}">
+    <isset property="env.IVY_HOME" />
+  </condition>
+  <property name="ivy.home" value="${user.home}/.ant" />
+  <property name="ivy.jar.dir" value="${ivy.home}/lib" />
+  <property name="ivy.jar.file" value="${ivy.jar.dir}/ivy.jar" />
+  <property name="reports.dir" value="${build.dir}/report"/>
 
   <!-- Use add-modules with Java 9 and higher. -->
   <condition property="java.modules"
       value="" else="--add-modules java.xml.bind">
     <matches string="${ant.java.version}" pattern="^1\.[678]$"/>
   </condition>
 
-  <path id="adaptorlib.build.classpath">
+  <path id="build.classpath">
```

Comment:
Indeed. My intention was to simplify the classpaths and reduce the number to compile,runtime,test, misc (for experimental / examples, etc), as it seemed to me that there is no real necessity of having them on a per-component scope..

I can rename them back.. however, can we take a look at the connector builds and see if this could be a problem, so i don't insist to change these, I just thought this makes it simpler.. Since these are internal references, I think it only could be an issue if a connector's ant scrip imports the library's build.xml, which sounds a bit weird to me, but let me check if this is the case..

Reply:
Hi szab100,

It looks to me like there are at least two too many intentions behind this CL.  

Smaller CLs are easier to review with higher confidence (less risk).  Smaller CLs can be rolled back individualls. 

I can imagine considering a CL that adds ivey, I can imagine a CL that consider coberatura, I can imagine a CL that considers suggests changing naming of classpaths, i can imagine CL that redoes eclipse.  

Is there really any benefit to doing thise changes together? I think the ivey CL could be OK given that ivey doesn't have to be downloaded seperatly, but what concrete problem are we fixing?

The classpaths are granular, what problem are we fixing?

What is the problem with eclipse?

And what is the problem with cobertura being out? 

I am open to discussing problems or making things better, but I would like us to have articulated objectives for each of the changes. And to have them seperate.

</can chat more later, gotta runt o meeting>

Reply:
Thank you PJ, yes let's discuss it after your meeting.

1) Smaller CLs are OK, except that it would leave the code broken between the consecutive changelists. However, I wouldn't split it up to more than 2 changelists (Ivy + Eclipse).

2) The classpath names are somewhat linked to and driven by the Ivy change. The general java standard / convention is to define clear and simple dependency & classpath namespaces: compile, runtime, test, (+extras, if necessary). I analyzed the classpath structure and it didn't really make sense to have separate classpath for each components, it added unnecessary complexity and many extra lines to the build file.. So simplifying them brings more clarity in my opinion..

I checked the connectors' sources and none of them imports the library's build file (fortunately, that would be super ugly), so these are internal references only and I would prefer the simplified version. Especially, if we want to introduce Ivy to them as well.

3) Cobertura is still out, meaning we don't include it with the sources. Adding it as an ivy dependency results in extra convenience, similar with ivy, users don't have to download it separately. Using it with the 'buildtools' ivy namespace clearly distinguishes it from the code, making it fully gpl-compliant.